### PR TITLE
Fix for issue #171

### PR DESCRIPTION
**Pull request by EricLiclair**
Am not sure about the variable 'gain', but I had a look at the line number 371 and implemented the necessary changes as mentioned in this comment https://github.com/fossasia/pslab-python/issues/171#issuecomment-787070650

I noticed a gap in the comments and thought if there might be some reason the issues is not solved yet.

Also, I do not have a PSLab device to record new serial traffic. Further help needed.

### CODE REVIEW

File: pslab/instrument/oscilloscope.py [hunk 1]
```diff
@@ -132,6 +132,10 @@ def capture(
 
         self._check_args(channels, samples, timegap)
         timegap = int(timegap * 8) / 8
+        
+        for channel in ("CH1", "CH2"):
+            self._set_gain(channel, gain)
```
Comment (bessman) [inline pslab/instrument/oscilloscope.py:137]: The gain value is stored in `AnalogInput` instances, which are in turn stored in the `_channels` attribute of the `Oscilloscope` class. You can access the stored gain values like this:
```python
gain = self._channels[channel].gain
```
Where `channel` is the name of the channel in question (CH1 or CH2) as a string.

File: pslab/instrument/oscilloscope.py [hunk 2]
```diff
@@ -132,6 +132,10 @@ def capture(
 
         self._check_args(channels, samples, timegap)
         timegap = int(timegap * 8) / 8
+
```
Comment (bessman) [inline pslab/instrument/oscilloscope.py:135]: Codacy is complaining about trailing white space on this line. Please remove it.

**Review by bessman — CHANGES_REQUESTED** [review]
Hi Eric, thanks for the pull request! You've got the right idea for how to fix the bug, please see my comment on line 137.

File: pslab/instrument/oscilloscope.py (hunk 2 above)
Comment (EricLiclair) [reply pslab/instrument/oscilloscope.py]: Done. Also, in CONTRIBUTING.md line 65 is a bit confusing. (😅 if you could add a comment to this thread just for me to understand that better. Thank You)

**Review by bessman — APPROVED** [review]

**Comment by bessman** [comment]
Thank you, your fix has been [merged](https://github.com/fossasia/pslab-python/commit/3f3e51912ed3e433d74af25bd891feae40824586) :smile: 
Since I had to record some serial traffic, I have merged your branch outside of this pull request. This pull request will therefore be closed.

**Pull request closed without merging**